File: Nian_Karl/QL/src/org/uva/ql/ast/expression/Expression.java
package org.uva.ql.ast.expression;

import org.uva.ql.ast.Node;
import org.uva.ql.ast.builder.CodePosition;
import org.uva.ql.ast.type.Type;
import org.uva.ql.typecheck.TypeChecker;

public abstract class Expression implements Node {
	private final CodePosition position;

	public Expression(CodePosition pos) {
		this.position = pos;
	}

	public CodePosition getPosition() {
		return position;
	}
	
	public abstract Type getType(TypeChecker typeChecker);
	
}

File: CarlaErik/UvA.SoftCon.Questionnaire/QL/AST/Model/Expressions/Expression.cs
using System.Collections.Generic;
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QL.AST.Model.Expressions
{
    public abstract class Expression : QLNode
    {
        protected Expression(TextPosition position)
            : base(position) { }

        public abstract DataType GetType(IDictionary<string, DataType> symbolTable);
    }
}
